pkg/commands/compute/computeacl/computeacl_test.go

```diff
@@ -0,0 +1,611 @@
+				Prefix: "192.168.0.0/16",
+				Action: "BLOCK",
+			},
+		},
+		Meta: computeacls.MetaEntries{
+			Limit: 100,
+		},
+	}
+
+	scenarios := []testutil.CLIScenario{
+		{
+			Name:      "validate missing --acl-id flag",
+			Args:      "",
+			WantError: "error parsing arguments: required flag --acl-id not provided",
+		},
+		{
+			Name: "validate bad request",
+			Args: "--acl-id bar",
+			Client: &http.Client{
+				Transport: &testutil.MockRoundTripper{
+					Response: &http.Response{
+						StatusCode: http.StatusBadRequest,
+						Status:     http.StatusText(http.StatusBadRequest),
+						Body: io.NopCloser(bytes.NewReader(testutil.GenJSON(`
+							{
+    							"title": "invalid ACL ID",
+    							"status": 400
+							}
+						`))),
+					},
+				},
+			},
+			WantError: "400 - Bad Request",
+		},
+		{
+			Name: "validate API success (empty list)",
+			Args: fmt.Sprintf("--acl-id %s", aclID),
+			Client: &http.Client{
+				Transport: &testutil.MockRoundTripper{
+					Response: &http.Response{
+						StatusCode: http.StatusOK,
+						Status:     http.StatusText(http.StatusOK),
+						Body: io.NopCloser(bytes.NewReader(testutil.GenJSON(computeacls.ComputeACLEntries{
+							Entries: []computeacls.ComputeACLEntry{},
+							Meta: computeacls.MetaEntries{
+								Limit: 100,
+							},
+						}))),
+					},
+				},
+			},
+			WantOutput: fmtComputeACLEntries(nil),
+		},
+		{
+			Name: "validate API success",
+			Args: fmt.Sprintf("--acl-id %s", aclID),
+			Client: &http.Client{
+				Transport: &testutil.MockRoundTripper{
+					Response: &http.Response{
+						StatusCode: http.StatusOK,
+						Status:     http.StatusText(http.StatusOK),
+						Body:       io.NopCloser(bytes.NewReader(testutil.GenJSON(entries))),
+					},
+				},
+			},
+			WantOutput: fmtComputeACLEntries(entries.Entries),
+		},
+		{
+			Name: "validate optional --json flag",
+			Args: fmt.Sprintf("--acl-id %s --json", aclID),
+			Client: &http.Client{
+				Transport: &testutil.MockRoundTripper{
+					Response: &http.Response{
+						StatusCode: http.StatusOK,
+						Status:     http.StatusText(http.StatusOK),
+						Body:       io.NopCloser(bytes.NewReader(testutil.GenJSON(entries))),
+					},
+				},
+			},
+			WantOutput: fstfmt.EncodeJSON(entries.Entries),
+		},
+	}
+
+	testutil.RunCLIScenarios(t, []string{root.CommandName, sub.CommandName, "list-entries"}, scenarios)
+}
+
+func fmtComputeACL(acl *computeacls.ComputeACL) string {
+	var b bytes.Buffer
+	text.PrintComputeACL(&b, "", acl)
+	return b.String()
+}
+
+func fmtComputeACLs(acls []computeacls.ComputeACL) string {
+	var b bytes.Buffer
+	text.PrintComputeACLsTbl(&b, acls)
+	return b.String()
+}
+
+func fmtComputeACLEntry(entry *computeacls.ComputeACLEntry) string {
+	var b bytes.Buffer
+	text.PrintComputeACLEntry(&b, "", entry)
+	return b.String()
+}
+
+func fmtComputeACLEntries(entries []computeacls.ComputeACLEntry) string {
+	var b bytes.Buffer
+	text.PrintComputeACLEntriesTbl(&b, entries)
+	return b.String()
+}
```

Comment:
I'm a little concerned about these functions, as they are using the same formatters as the code-under-test uses, so that means the code-under-test is not being treated as a 'black box'.

In specific, as noted in my other comment the 'Action' is not being included in the output from `PrintComputeACLEntry` when a single entry is output (due to the lack of flushing the `PrefixWriter`), but the test is not failing because the test uses the same formatter.

In other words: the tests should probably be using their own formatters, or have hardcoded 'golden' strings to compare against.

Reply:
An alternative solution is to leave these functions as they are, but add unit tests for the `PrintComputeACL...` functions themselves. Those units tests would have hardcoded input and output for comparison, and they would ensure that all of the provided input appears in the output and in the correct format.

Reply:
Addressed in e7fe55a0ba6586ae4ddae940b05ed79d65d3ccc4.